### PR TITLE
Add depth rendering from camera

### PR DESCRIPTION
**Pull request by 23pointsNorth**
Allow the mujoco renderer to return the depth image using the `depth_array` mode. 

It follows the `rgb_array` extraction method.

### CODE REVIEW

**Comment by pzhokhov** [comment]
looks good but, could you comment more on why is that needed?

**Comment by 23pointsNorth** [comment]
As part of my research I am working on adding visual env for the mujoco [robotics](https://gym.openai.com/envs/#robotics) env. 
Variations of those would include rgb/d/rgbd data as input. 

Given the trend of applying RL for robotics, we should add more of the tools used in that community.

File: gym/envs/mujoco/mujoco_env.py [hunk 1]
```diff
@@ -112,6 +112,13 @@ def render(self, mode='human', width=DEFAULT_SIZE, height=DEFAULT_SIZE):
             data = self._get_viewer(mode).read_pixels(width, height, depth=False)
             # original image is upside-down, so flip it
             return data[::-1, :, :]
+        elif mode == 'depth_array':
+            self._get_viewer(mode).render(width, height)
+            # window size used for old mujoco-py:
+            # Extract depth part of the read_pixels() tuple
+            data = self._get_viewer(mode).read_pixels(width, height, depth=True)[1]
+            # original image is upside-down, so flip it
+            return data[::-1, :]
```
Comment (pzhokhov) [inline gym/envs/mujoco/mujoco_env.py]: actually, you lost me now... Why is depth data 2D? Do we still need to flip it upside down?

Comment (23pointsNorth) [reply gym/envs/mujoco/mujoco_env.py]: RGB is [h, w, 3] where 3 is for the r, g, b.
Depth I initially assumed is [h, w, 1] , after local testing, it turned out to be [h, w].

Comment (pzhokhov) [reply gym/envs/mujoco/mujoco_env.py]: oh got it. Ok.

**Pull request merged**